<?xml version="1.0"?>
<launch>
  <!-- lidar filtering arguments -->
  <arg name="laser_topic" default="velodyne_points" />
  <arg name="scan_line" default="32" />
  <arg name="max_distance" default="130.0" />
  <arg name="min_distance" default="5.0" />
  <arg name="vertical_resolution" default="0.33" />
  <arg name="filtering_leaf_size" default="0.2" />
    <!-- <arg name="pcd_map_name" default="dlo_map.pcd" /> -->
  <arg name="robot_x_min" default="-2.0" />
  <arg name="robot_x_max" default="2.0" />
  <arg name="robot_y_min" default="-0.1" />
  <arg name="robot_y_max" default="0.1" />
  <arg name="robot_z_min" default="-1.0" />
  <arg name="robot_z_max" default="1.0" />
  
  <!-- rviz -->
  <arg name="rviz" default="true" />
  
  <!-- if using bagfile, set use_sim_time true-->
  <param name="/use_sim_time" value="true" />
  
  <!-- lidar filtering parameters -->
  <param name="laser_topic" type="string" value="$(arg laser_topic)" />
  <param name="scan_line" type="int" value="$(arg scan_line)" />
  <param name="max_distance" type="double" value="$(arg max_distance)" />
  <param name="min_distance" type="double" value="$(arg min_distance)" />
  <param name="vertical_angle" type="double" value="$(arg vertical_resolution)" />
  <param name="filtering_leaf_size" type="double" value="$(arg filtering_leaf_size)" />
  <param name="robot_x_min" type="double" value="$(arg robot_x_min)" />
  <param name="robot_x_max" type="double" value="$(arg robot_x_max)" />
  <param name="robot_y_min" type="double" value="$(arg robot_y_min)" />
  <param name="robot_y_max" type="double" value="$(arg robot_y_max)" />
  <param name="robot_z_min" type="double" value="$(arg robot_z_min)" />
  <param name="robot_z_max" type="double" value="$(arg robot_z_max)" />

  <!-- node -->
  <node pkg="velodyne_ndt" type="lidar_processing_node" name="lidar_processing_node" output="screen">
  </node>

  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find velodyne_ndt)/rviz/filtered_point_rviz.rviz" />
  </group>

</launch>
